lib/sig0.js

```diff
@@ -97,6 +97,18 @@ sig0.createKey = function createKey(algorithm, publicKey) {
   rd.flags = 0;
   rd.protocol = 0;
   rd.algorithm = algorithm;
+
+  // https://tools.ietf.org/html/rfc4034#appendix-A.1.1
+  // PRIVATEDNS publicKey should be wire encoded and prefixed with algo name
+  if (algorithm === sig0.algs.PRIVATEDNS) {
+    const name = "secp256k1sha256";
+    const key = bio.write(1 + name.length + 1 + publicKey.length);
+    key.writeU8(name.length);
```

Comment:
Should we be worried about overflows here? 255 is the max U8 and also the max size of a domain name according to the spec:

```
To simplify implementations, the total length of a
domain name (i.e., label octets and label length octets)
is restricted to 255 octets or less.
```
https://stackoverflow.com/questions/32290167/what-is-the-maximum-length-of-a-dns-name

Reply:
Yes, this should be a hard limit. We could add an `assert`.